### DISCUSSION TITLE
A tutorial for UERANSIM & Open5GS on Ubuntu - VirtualBox

### DISCUSSION THREAD

**Discussion by mohitkr05** [aligungr/UERANSIM#305] [General]
Hello,

I have created a tutorial video for installation of Open5GS + UERANSIM on Ubuntu , VirtualBox and thought it would be useful for reference.

https://www.youtube.com/watch?v=DxhS2wdbMtc

**Comment by aligungr** [comment]
Thanks Mohit, I put the link in the UERANSIM Wiki page.

**Comment by opentelcokloud** [comment]
Thank you for your sharing.
I just also try to implement the (Containerized) Open5gs with srs-RAN to set up complete 4G LTE network up and running on my Native Cloud Kubernetes Cluster with some good results. Please join me @all if you are interested in this

https://www.youtube.com/channel/UCdrZzzEgfXbsPfnvovHez9w

**Reply by aligungr** [reply]
@opentelcokloud

Thanks for sharing the video. It could be helpful for other people. (But I cannot share the video herein since it's unrelated to UERANSIM)

Cheers

**Reply by opentelcokloud** [reply]
Thanks Aligungr.
I also used UERANSIM with Open5gs. It’s great.
Will try to share some results about that too.

Thank you!

**Reply by kodesam** [reply]
any one  can share installation of Open5GS on kubernetes + UERANSIM on Ubuntu VM . 

I Have installed but getting SCTP connection error

UERANSIM/build$ ./nr-gnb -c ../config/open5gs-gnb.yaml
UERANSIM v3.2.2
[2021-07-12 02:15:35.040] [sctp] [info] Trying to establish SCTP connection... (10.111.240.250:38412)
[2021-07-12 02:20:39.639] [sctp] [error] Connecting to 10.111.240.250:38412 failed. SCTP could not connect: Connection timed out

**Reply by infinitydon** [reply]
The following issue on open5gs might help:

https://github.com/open5gs/open5gs/issues/1060

**Reply by opentelcokloud** [reply]
@KODESAM: 
gNB/UE VM could not reach MME/EPC (Pods) as they are in different network segments.
1 quick solution for this is you add a STATIC ROUTE at the UE VM to the Kubernetes Pod's network segment.
Lets say: If your Kubernetes's Pod Network is (A):10.111.0.0/16; The host IP Address of the Kubernetes cluster's host (that the MME pod is now in) is (B): 192.168.0.10.
Then you add the route that have the network of (A) and gateway of (B) into your UE VM, that would help.

I suppose in real life network, 1 external router (or kind of) would be implemented
